libs/wingsdk/src/ex/react-website.md

```diff
@@ -0,0 +1,86 @@
+---
+title: React Website
+id: react-website
+description: A built-in resource for creating deployable websites using the React framework.
+keywords: [Website, React, deployment, build]
+sidebar_position: 1
+---
+
+The `ex.ReactWebsite` resource represents a website, built using React, that can be both hosted in the cloud or to run on a development hot-reload server in the sim target.
+
+## Usage
+
+### Initialization
+
+Using the default arguments:
+
+```ts
+bring cloud;
+bring util;
+
+let website = new cloud.Website(projectPath: "./client", isDevRun: util.tryEnv("ENV") == "dev");
+```
+
+or customizing them:
+
+```ts
+bring cloud;
+bring util;
+
+let website = new cloud.Website(
+  projectPath: "./client",
+  isDevRun: util.tryEnv("ENV") == "dev"
+  buildFolder: "/dist" // default is "/build"
+  startCommand: "pnpm start" // default is "npm start"
+  buildCommand: "pnpm build" // default is "npm build"
+  hostProps: {} // website resource props (expect of the "path" prop)
+  localPort: "4000" // default is 3001
+ );
+```
+
+- When calling `wing it` when `isDevRun` is `true`- wing starts both the simulator and React's development server
+- When calling `wing it`/`wing compile` on the sim target, when `isDevRun` is `false`- wing builds React and serves it over the [Website resource](./website.md).
+- When calling `wing compile` to any other target - wing builds react locally then deploys it to the cloud! (using the [Website resource](./website.md))
```

Comment:
This behavior is a little bit strange - usually `wing compile` is not deploying anything. I understand you're probably doing that because you want to support hot-reloading, but it is surprising that `wing compile` behaves differently here. Perhaps we should introduce `wing watch`?

Reply:
No, it doesn't actually deploys, I'll change the working, thanks for noticing!